hazelcast/src/main/java/com/hazelcast/internal/config/override/PropertiesToNodeConverter.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.config.override;
+
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+final class PropertiesToNodeConverter {
+
+    private PropertiesToNodeConverter() {
+    }
+
+    static ConfigNode propsToNode(Map<String, String> properties) {
+        String rootNode = findRootNode(properties);
+
+        ConfigNode root = new ConfigNode(rootNode);
+        for (Map.Entry<String, String> e : properties.entrySet()) {
+            parseEntry(new AbstractMap.SimpleEntry<>(e.getKey().replaceFirst(rootNode + ".", ""), e.getValue()), root);
+        }
+        return root;
+    }
+
+    private static String findRootNode(Map<String, String> properties) {
+        Set<String> rootNodeNames = properties.keySet().stream()
+          .map(key -> key.split("\\.")[0])
+          .collect(Collectors.toSet());
+
+        if (rootNodeNames.size() > 1) {
+            throw new IllegalStateException("parsed config entries have conflicting root node names");
```

Comment:
`InvalidConfigurationException`? Same in rest of class. Also, maybe you can include the conflicting names.